client/src/SubscriptionDetailsCard.js

```diff
@@ -0,0 +1,65 @@
+import "./SubscriptionDetailsCard.css";
+import { useState, useEffect } from "react";
+import CircleCheck from "./CircleCheck";
+
+const SUBSCRIPTION_INTERVALS = {
+    month: "monthly",
+    year: "yearly",
+};
+
+const SubscriptionDetailsCard = ({
+    id,
+    selectedCardId,
+    subscriptionName="Add name here",
+    price=0,
+    interval=SUBSCRIPTION_INTERVALS.month,
+    subscriptionDetails=["Add description details here"]
+}) => {
+    const [hovered, setHovered] = useState(false);
+    const [selected, setSelected] = useState(false);
+
+    const capitalize = (str) => str.charAt(0).toUpperCase() + str.slice(1);
+    const determineCardWasSelected = () => {
+        if(id !== undefined && selectedCardId !== undefined){
+            if(id === selectedCardId){ setSelected(true) }
+        }
+    };
+
+    useEffect(() => {
+        determineCardWasSelected()
+        if(id !== selectedCardId){ setSelected(false) }
```

Comment:
can we move this line of logic into the function above `determineCardWasSelected()`, so that the logic remains together?

Reply:
Initially I did try putting into the determineCardWasSeclected. I decided to keep it outside because it does not behave in the same way. I believe it's because it's trying to run the if statement at the same time. Going to keep it as is since I know it works this way and we're in a time crunch atm, I will address it in a follow up PR.